src/EquationGen.js

```diff
@@ -388,11 +388,19 @@ export default class EquationGen extends ModelReader {
       // If the data was found, convert it to a lookup.
       if (getCellValue) {
         let indexNum = 0
-        if (!R.isEmpty(this.var.subscripts)) {
+        if (!R.isEmpty(this.var.separationDims)) {
           // Generate a lookup for a separated index in the variable's dimension.
-          // TODO allow the index to be in either position of a 2D subscript
-          let ind = sub(this.var.subscripts[0])
-          indexNum = ind.value
+          if (this.var.separationDims.length > 1) {
+            console.error(`WARNING: direct data variable ${this.var.varName} separated on more than one dimension`)
```

Comment:
This is not a blocking comment because I think we have other existing cases like this (treating unexpected cases as a soft warning), but I wonder if we should make this and similar cases a hard error (exit with non-zero code).  At some point it might be good to revisit error handling in SDE to be a bit more fail-fast and consistent with error reporting mechanisms, but that's for another day.